src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -507,6 +519,36 @@ byte[] getEncodedInternal() {
         }
     }
 
+    private static class P11PBEKey extends P11SecretKey implements PBEKey {
```

Comment:
Yes, if it needs to be extended, it would be in the context of `P11Key.java` modifications (since it is `private`), so `final` can be removed in that unlikely hypothetical case.

I will make it `final`.